File: shoring/blackjackEngine/src/test/java/com/panopset/tests/blackjackEngine/HxTest.java
package com.panopset.tests.blackjackEngine;

import static com.panopset.blackjackEngine.CommandDefinitions.CMD_DEAL;
import static com.panopset.blackjackEngine.CommandDefinitions.CMD_DOUBLE;
import static com.panopset.blackjackEngine.CommandDefinitions.CMD_HIT;
import static com.panopset.blackjackEngine.CommandDefinitions.CMD_STAND;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import com.panopset.blackjackEngine.BlackjackConfigDefault;
import com.panopset.blackjackEngine.BlackjackGameEngine;

public class HxTest extends AbstractBlackjackTest {

  @Test
  void testSingleDeck() {
    BlackjackGameEngine bge = new BlackjackGameEngine(new BlackjackConfigDefault() {
      @Override
      public int getDecks() {
        return 1;
      }
    });
    verifyRecommendedActions(bge, new String[] {CMD_DEAL, CMD_DOUBLE}, soft18v2());
    performDeviantActions(bge, new String[] {CMD_DEAL, CMD_HIT, CMD_HIT}, soft18v2());
    Assertions.assertEquals("Busted",
        bge.getCycle().getPlayers().get(0).getHands().get(0).getMessage());
  }

  @Test
  void test() {
    BlackjackGameEngine bge = new BlackjackGameEngine(new BlackjackConfigDefault() {
      @Override
      public int getDecks() {
        return 6;
      }
    });
    verifyRecommendedActions(bge, new String[] {CMD_DEAL, CMD_HIT, CMD_STAND}, soft18v2());
    performDeviantActions(bge, new String[] {CMD_DEAL, CMD_HIT, CMD_HIT}, soft18v2());
    Assertions.assertEquals("Busted",
        bge.getCycle().getPlayers().get(0).getHands().get(0).getMessage());
  }
}

File: code/shoring/blackjackEngine/src/test/kotlin/com/panopset/tests/blackjackEngine/HxTest.kt
package com.panopset.tests.blackjackEngine

import com.panopset.blackjackEngine.*
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class HxTest {
    @Test
    fun testSingleDeck() {
        val bge = BlackjackGameEngine(BlackjackConfigSingleDeckTest)
        verifyRecommendedActions(bge, arrayOf(CMD_DEAL, CMD_DOUBLE), soft18v2())
        performDeviantActions(bge, arrayOf(CMD_DEAL, CMD_HIT, CMD_HIT), soft18v2())
        Assertions.assertEquals(
            "Busted",
            bge.getCycle().players[0].hands[0].message
        )
    }

    @Test
    fun test() {
        val bge = BlackjackGameEngine(object : BlackjackConfigBaseTest() {
            override fun getDecks(): Int {
                return 6
            }
        })
        verifyRecommendedActions(bge, arrayOf(CMD_DEAL, CMD_HIT, CMD_STAND), soft18v2())
        performDeviantActions(bge, arrayOf(CMD_DEAL, CMD_HIT, CMD_HIT), soft18v2())
        Assertions.assertEquals(
            "Busted",
            bge.getCycle().players[0].hands[0].message
        )
    }
}
